### PR TITLE
Enhancements #240, #233 Comments reply and other actions

### PR DESCRIPTION
**Pull request by dellagustin**
This commit implements #240 and initial support for #233.

A menu is added to comments that support initiating the following actions:
- Reply to post
- Copy link to post
- Open in original site - opens the post in its original system at another tab
- Follow author of the post
- Forget interactor account

This commit is based on
https://github.com/nathangathright/comment-thread-mockup/commit/6b2f71628f359a6f4cfadcab47c301b85a976b58

## Known issues

- When the user is first prompted to enter their account, the opening of a new tab for interaction will be blocked as a pop-up.
- ~When using the dark mode, the text in the menu does not appear (it is the same color as the background)~ - fixed with https://github.com/Podcastindex-org/web-ui/pull/253/commits/a0bd2770146f93f3a1c0ad8922d473462c072b4b and https://github.com/Podcastindex-org/web-ui/pull/251
- ~When you click outside of the menu, it does not disappear, it only disappear when you click a menu item or the menu button~ - fixed with https://github.com/Podcastindex-org/web-ui/pull/253/commits/851afe109009b4725551f97588e6efd7b6945419
- There are very little guardrails, so some might need to be implemented later (e.g. verifying the input for the interactor account).

## Open Questions
Which fediverse software (e.g. Mastodon, Pleroma) supports the remote interaction flow we are using here? At least Mastodon and Pleroma (actually Akkoma) were tested.

## Related issues

- https://github.com/Podcastindex-org/web-ui/issues/240
- https://github.com/Podcastindex-org/web-ui/issues/233

## Related links and references

- [OStatus 1.0 Draft 2](https://ostatus.github.io/spec/OStatus%201.0%20Draft%202.html#anchor10) - Section 11 - Discover
- https://socialhub.activitypub.rocks/t/represent-endpoint-for-remote-interaction/480
- WebFinger - https://www.rfc-editor.org/rfc/rfc7033
- https://www.hughrundle.net/how-to-implement-remote-following-for-your-activitypub-project/

## Implementation Details

The remote interaction is implemented using the WebFinger protocol/API and also the OStatus Subscribe mechanism.

In summary, it gets to remote interact URL template using WebFinger with a call to `https://<follower instance host>/.well-known/webfinger?resource=acct:<follower account>`, and inspect the response for the link with rel equals to `http://ostatus.org/schema/1.0/subscribe`, then builds the remote interact URL and opens it in a new tab.

### CODE REVIEW

**Comment by dellagustin** [comment]
## Demo

image
image
Known issue:
image

either open the pop up, or interact again.
@daveajones 's post is open on my instance (a login flow is triggered if I'm not already logged in):
image

**Comment by dellagustin** [comment]
@daveajones , ready for review, if you can leave with the known issues.
@nathangathright I was too lazy to implement a modal dialog, to avoid the pop up block, but since it only happens when you first enter your account, that was acceptable for me
@glaszig , @stevencrader , @nathangathright , it would be great if you could propose a fix to the known issue related to the "Dark mode" (related, https://github.com/Podcastindex-org/web-ui/pull/221, https://github.com/Podcastindex-org/web-ui/pull/251).

**Comment by daveajones** [comment]
10-4. Will review!

File: ui/src/components/Comments/index.tsx [hunk 1]
```diff
@@ -53,10 +65,21 @@ class Comment extends React.PureComponent<ICommentProps> {
                             <span className='handle'>{this.props.comment.attributedTo.account}</span>
                         </div>
                     </a>
-                    <span aria-hidden="true">·</span>
                     <a href={this.props.comment.url} className='permalink'>
                         <time>{this.props.comment.publishedAt.toLocaleString()}</time>
                     </a>
+                    <button className="context-menu" aria-label="More" onClick={() => this.onClickShowMenu()}>
+                        <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" width="24" height="24" stroke-width="1.5" stroke="currentColor">
+                            <path stroke-linecap="round" stroke-linejoin="round" d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z" />
```
Comment (stevencrader) [inline ui/src/components/Comments/index.tsx:73]: I get a warning in the console about `stroke-width`, `stroke-linecap`, and `stroke-linejoin`. According to PyCharm, they should be `strokeWidth`, `strokeLinecap`, and `strokeLinejoin`.

**Comment by stevencrader** [comment]
> When you click outside of the menu, it does not disappear, it only disappear when you click a menu item or the menu button

To solve this, you could look at the fix I added for the top bar menu on mobile. See `handleClickOutside` in: https://github.com/Podcastindex-org/web-ui/blob/b7d2b766b81ad03e7be69a39b6eea0f029ad3449/ui/src/components/TopBar/index.tsx#L57-L66

Since your menu will be removed when navigating unlike the top bar, you could get away with a much shorter timeout value. I found a value of 10 works.

File: ui/src/components/Comments/index.tsx (hunk 1 above)
Comment (dellagustin) [reply ui/src/components/Comments/index.tsx:73]: Thank you @stevencrader , I'll replace that.
Do you know that you can suggest changes in GitHub like this?
```suggestion
                        <svg xmlns="http://www.w3.org/2000/svg" fill="none" viewBox="0 0 24 24" width="24" height="24" strokeWidth="1.5" stroke="currentColor">
                            <path strokeLinecap="round" strokeLinejoin="round" d="M6.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM12.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0zM18.75 12a.75.75 0 11-1.5 0 .75.75 0 011.5 0z" />
```

**Comment by dellagustin** [comment]
Hello @stevencrader , thanks for tip with https://github.com/Podcastindex-org/web-ui/pull/253#issuecomment-1535789250.
I implemented it and it worked. It also worked without `setTimeout` entirely. Do you see any reason to keep it in my case?

File: ui/src/components/Comments/index.tsx (hunk 1 above)
Comment (dellagustin) [reply ui/src/components/Comments/index.tsx]: Resolved with https://github.com/Podcastindex-org/web-ui/pull/253/commits/09396eb87e42ece3d05413e265c7af83c897e26e.

**Comment by dellagustin** [comment]
@daveajones I added some fixed to the known issues with the help of @stevencrader .
From my side, this is ready to go.

File: ui/src/components/Comments/index.tsx (hunk 1 above)
Comment (stevencrader) [reply ui/src/components/Comments/index.tsx]: I didn't know about that. Thanks for the tip.

**Comment by stevencrader** [comment]
> Hello @stevencrader , thanks for tip with [#253 (comment)](https://github.com/Podcastindex-org/web-ui/pull/253#issuecomment-1535789250). I implemented it and it worked. It also worked without `setTimeout` entirely. Do you see any reason to keep it in my case?

I don't see any reason to keep the timeout if it isn't needed.
